zaza/utilities/machine_os.py

```diff
@@ -133,6 +133,9 @@ def _set_vfio_unsafe_noiommu_mode(unit, enable, model_name=None):
     :type model_name: Optional[str]
     :raises: AssertionError, zaza.model.CommandRunFailed
     """
+    zaza.utilities.juju.remote_run(
+        unit.name, 'modprobe vfio', model_name=model_name, fatal=True)
```

Comment:
Would an alternative be to use the already existing `load_kernel_module()` function for this?:
https://github.com/openstack-charmers/zaza/blob/51c90dbde87824d0cabdbb25cc85c418dbbc86c6/zaza/utilities/machine_os.py#L38-L54

And does the call belong here or on the test code (have no strong opinions on this one, just raising the question).

Reply:
Oh, good point about the `load_kernel_module`, I missed that.
Regarding the placement, I think that this is the right place. Up until now the tests called `enable_vfio_unsafe_noiommu_mode` with the assumption that the VFIO module is loaded. Loading the kernel here ensures that the assumption is correct (+ there are no side effects to trying to load already loaded module).